app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -41,38 +45,99 @@ export default class SchoolDisciplineDashboard extends React.Component {
+      if (selectedCategory && selectedChart === 'race' && student.race !== selectedCategory) return false;
+      if (selectedCategory && selectedChart === 'homeroom_label' && student.homeroom_label !== selectedCategory) return false;
+      if (selectedCategory && !this.filterIncidents(student.discipline_incidents, true).length) return false;
+      return true;
     });
-    return studentDisciplineIncidentCounts;
   }
 
-  sortChartKeys(groupedIncidents) {
-    const chartKeys = Object.keys(groupedIncidents);
-    switch(this.state.selectedChart) {
-    case 'time': return this.sortedTimes(chartKeys);
-    case 'day': return this.sortedDays(chartKeys);
-    case 'grade': return this.sortedGrades(chartKeys);
-    default: return this.sortedByIncidents(groupedIncidents); //because number of incidents in each category needed here
+  timeStampToHour(incident) {
+    return incident.has_exact_time ? moment.utc(incident.occurred_at).startOf('hour').format('h:mm a') : "Not Logged";
+  }
+
+  timeStampToDay(incident) {
+    return moment.utc(incident.occurred_at).format("ddd");
+  }
+
+  getIncidentsFromStudents(students) {
+    //for chart data, does not filter on selected categories
+    return _.flatten(students.map(student => this.filterIncidents(student.discipline_incidents, false)));
+  }
+
+  //Depending on the chart, incidents are grouped either by student attribute or incident attribute. Any future
+  //charts may be handled here.
+  getChartData(students, group) {
+    switch(group) {
+    case 'homeroom_label': case 'race': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedByStudent(groupedStudents);
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'grade': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedGrades(Object.keys(groupedStudents));
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'incident_code': case 'incident_location': {
+      const incidents = this.getIncidentsFromStudents(students);
+      const groupedIncidents = _.groupBy(incidents, group);
+      const categories = this.sortedByIncidents(groupedIncidents);
+      const seriesData = categories.map(category => groupedIncidents[category].length);
+      return {categories, seriesData};
+    }
+    case 'time': {
+      const incidents = this.getIncidentsFromStudents(students);
+      const groupedIncidents = _.groupBy(incidents, incident => {
+        return incident.has_exact_time ? moment.utc(incident.occurred_at).startOf('hour').format('h:mm a') : "Not Logged";
+      });
+      const categories = this.sortedTimes(Object.keys(groupedIncidents));
```

Comment:
I think there's a UX bug here here, in that for breakdowns that refer to stable known categories, we should set these values to be fixed so this looks like the scale people will expect (eg, like with `sortedDays` here rather than inferred from the data in `sortedTimes`).

This is what I see locally, check out the x scale:
<img width="759" alt="screen shot 2018-10-02 at 10 09 31 am" src="https://user-images.githubusercontent.com/1056957/46353918-4ea30500-c62b-11e8-85fe-98b5daec84e2.png">

Reply:
I'd imagine this looks less awkward with non-random data, but what do you think an appropriate scale would be? It should be possible to display incidents outside of school hours, but a 24 hour scale feels unnecessary. Maybe I'll try it with full 24 hours and we can see if it looks okay in production.

Reply:
Yeah I agree that there's probably not a lot happening in most schools at 1am :)  Maybe 7am - 3pm and bucket anything else into "other"?